Source/WebCore/accessibility/isolatedtree/AXIsolatedTree.h

```diff
@@ -491,6 +492,9 @@ class AXIsolatedTree : public ThreadSafeRefCountedAndCanMakeThreadSafeWeakPtr<AX
     // as the original parent will want to queue it for removal, but we need to keep the object around
     // for the new parent.
     HashSet<AXID> m_protectedFromDeletionIDs;
+    // True if m_protectedFromDeletionIDs has changed, and m_pendingProtectedFromDeletionIDs needs to be updated as a result.
+    // Only accessed on the main-thread.
+    bool m_protectedFromDeletionIDsIsDirty { false };
     // Only accessed on the main thread.
     // Objects whose parent has changed, and said change needs to be synced to the secondary thread.
     HashSet<AXID> m_needsParentUpdate;
```

Comment:
This could be greatly simplified if we move away from calling applyPendingChanges for every AX thread request, but instead we update the iso tree on the main thread holding the lock. The problem of serving an AX thread request during an update should disappear altogether.

Reply:
Seems like a good follow-up change.